src/Platforms.jl

```diff
@@ -87,7 +87,7 @@ const ARCHITECTURE_FLAGS = Dict(
     # Many compiler flags are the same across clang and gcc, store those in "common"
     "common" => Dict(
         "i686" => Dict(
-            "i686" => ["-march=i686", "-mtune=generic"],
+            "i686" => ["-march=pentium4", "-mtune=generic"],
```

Comment:
```suggestion
            "pentium4" => ["-march=pentium4", "-mtune=generic"],
```

Let's just call it `pentium4` everywhere.

Reply:
Huh, OK... but wouldn't that requires lots of changes here and also elsewhere? And have a risk of breaking things in all kinds  of subtle and unsubtle ways? I really don't dare make such a change myself, as I don't understand the ramifications and the big picture, so I'd rather prefer if somebody else who feels more confident in this took care of it.

Reply:
Sorry, I didn't mean changing the architecture family name; just this "base microarchitectural level".

In reality, now that [this Base PR](https://github.com/JuliaLang/julia/pull/38186) was merged, I believe the only other thing that needs to change is [the docstring here](https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/master/src/auditor/instruction_set.jl#L81)

Reply:
And since that change is in `BinaryBuilder.jl` I think this PR can be merged after it goes green.